File: Problems/No.1566_Detect_Pattern_of_length_M_Repeated_K_or_More_Times/detect_pattern_of_length_m_repeated_k_or_more_times.py
class Solution(object):
    def containsPattern(self, arr, m, k):
        """
        :type arr: List[int]
        :type m: int
        :type k: int
        :rtype: bool
        """

        # parameter
        pattern_cnt = 0

        # start computing
        if (len(arr) < m*k):
            return False
        for i in range(len(arr)-m):
            if (arr[i] == arr[i+m]):
                pattern_cnt += 1
            else:
                pattern_cnt = 0
            
            if (pattern_cnt == m*(k-1)):
                return True
        
        return False

File: Problems/No.1566_Detect_Pattern_of_length_M_Repeated_K_or_More_Times/detect_pattern_of_length_m_repeated_k_or_more_times.cpp
class Solution {
public:
    bool containsPattern(vector<int>& arr, int m, int k) {
        
        // parameter
        int pattern_cnt = 0;

        // start computing
        if (arr.size() < m*k) return false;
        for (int i=0; i<arr.size()-m; i++) {

            if (arr[i] == arr[i+m]) pattern_cnt++;
            else pattern_cnt = 0;

            if (pattern_cnt == m*(k-1)) return true;

        }

        return false;

    }
};
